src/test/java/com/yegor256/tojos/MnSynchronizedTest.java

```diff
@@ -0,0 +1,186 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.yegor256.tojos;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test case for {@link MnSynchronized}.
+ *
+ * @since 0.3.0
+ */
+class MnSynchronizedTest {
+
+    /**
+     * Number of threads.
+     */
+    static final int THREADS = 5;
+
+    /**
+     * The number of changes in under-test mono.
+     */
+    private final AtomicInteger counter = new AtomicInteger(0);
+
+    /**
+     * The mono under test.
+     */
+    private Mono shared;
+
+    /**
+     * The executor.
+     */
+    private ExecutorService executor;
+
+    @BeforeEach
+    final void setUp() {
+        this.shared = new MnSynchronized(new MnMemory());
+        this.executor = Executors.newFixedThreadPool(MnSynchronizedTest.THREADS);
+    }
+
+    /**
+     * Thread-safety test.
+     * In this test, we check the number of changes in MnSynchronized mono.
+     * It should be equal to the sum of the arithmetic progression over the number of threads.
+     *
+     * @throws InterruptedException When fails
+     */
+    @Test
+    final void concurrentScenario() throws InterruptedException {
+        for (int trds = 1; trds <= MnSynchronizedTest.THREADS; ++trds) {
+            this.executor.submit(
+                new MnSynchronizedTest.TestTask(
+                    MnSynchronizedTest.rowsBySize(trds),
+                    this.shared,
+                    this.counter
+                )
+            );
+        }
+        this.executor.shutdown();
+        assert this.executor.awaitTermination(5L, TimeUnit.SECONDS);
+        MatcherAssert.assertThat(
```

Comment:
@l3r8yJ  I don't see the reason to check `this.counter` value. The counter has the type `AtomicInteger` which purpose is to provide thread-save access to a variable. In other words, you check how `AtomicInteger` works, but it tells nothing about `MnSynchronized` thread-safe characteristics.